connexion/apis/aiohttp_api.py

```diff
@@ -293,13 +293,53 @@ async def get_request(cls, req):
         :rtype: ConnexionRequest
         """
         url = str(req.url)
-        logger.debug('Getting data and status code',
-                     extra={'has_body': req.has_body, 'url': url})
+
+        logger.debug(
+            'Getting data and status code',
+            extra={
+                # has_body | can_read_body report if
+                # body has been read or not
+                # body_exists refers to underlying stream of data
+                'body_exists': req.body_exists,
+                'can_read_body': req.can_read_body,
+                'content_type': req.content_type,
+                'url': url,
+            },
+        )
 
         query = parse_qs(req.rel_url.query_string)
         headers = req.headers
         body = None
-        if req.body_exists:
+
+        # if request is not multipart, `data` will be empty dict
+        # and stream will not be consumed
+        post_data = await req.post()
```

Comment:
@ddurham2 
I this approach may be problematic.

Referencing this doc:
https://docs.aiohttp.org/en/stable/web_quickstart.html?highlight=multipart#file-uploads
And the source code here:
https://github.com/aio-libs/aiohttp/blob/v3.7.4/aiohttp/web_request.py#L639

The above code comment seems to be assuming that the stream will not be consumed, but as per the above source code it does read the entire multipart request into memory. 

What I was finding, was that running something like 
```
multipart = await request.multipart()
part = await multipart.next()
```

results in the following error:
```
ValueError: Could not find starting boundary b'--94aa6f10ca444684a68e350816dbb465'
```

This is because connexion already reads and drains the stream before it gets to my handler. The `File Uploads` link above points out that file uploads should be handled using `request.multipart` to avoid OOM errors.

I'm not familiar with the rest of connexion yet to suggest solution, but I wanted to raise it here as we've been troubleshooting why `request.multipart` was not working.

Reply:
Thanks for your feedback.

I believe what I meant by that comment was that if it's not multipart, then getting the value of post() will result in the post_data variable being empty (I should correct the comment to say 'post_data') and that it won't consume the request stream.  
However, looking at the code ( https://github.com/aio-libs/aiohttp/blob/v3.7.4/aiohttp/web_request.py#L629 ) that should actually say if it's not 'application/x-www-form-urlencoded' nor 'application/x-www-form-urlencoded' then it will be empty... 
I will clarify the comment, and it appears that my change will fix x-www-form-encoded forms as well, which, it appears, previously wouldn't extract those values from the request.

On the concern about it reading into memory, it actually won't necessarily read and hold the entire request data in memory.  My main concern there would be uploaded files (as seems to be your concern too), being too large to fit in memory.  But as you can see in https://github.com/aio-libs/aiohttp/blob/v3.7.4/aiohttp/web_request.py#L656 , it efficiently streams any files in the request from the network into temp files.
However, any non-files values will be momentarily read into memory (one at a time) but will discard them if they are larger than a max.  See https://github.com/aio-libs/aiohttp/blob/v3.7.4/aiohttp/web_request.py#L663  aiohttp could perhaps do a better job at reading only up to the maximum acceptable field size into memory rather than reading the whole thing in before rejecting it.  But that's an enhancement for them to make.

Now, to your point about not being able to read the data from the request yourself... Yes, it means that if it's multipart (or x-www-form-encoded forms), then the request object passed to an application's handler cannot be used to read the request a second time using multipart().  However, `await post()` should still work and return a dict with the form values.
Is using the request object for this purpose something you _needed_ to do, or just something you tried to do (with my patch in place) due to the assumed memory issues?

Reply:
So, I reverted back to the original behavior for _get_body_argument() which no longer breaks body parameters into handler arguments.

However, in the process I learned, or.. remembered (this original change being so long ago now).. that the reason I though body should be split out is because uploaded files in the body _are_ broken out as individual handler arguments 
 which is an existing behavior apparently.

My main change here was to make the aiohttp handler consistent with flask.  I did that by constructing the ConnexionRequest() object in aiohttp_api.py the same as flask does regarding files.

So, I'm just pointing out that some future cleanup might want to make the file elements stay inside of body, where I'd think you'd say they belong due to the reasoning that they're not top-level parameters.  But I think it doesn't work that way because this is basically flask's existing behavior.

Reply:
Hopefully, we're ready to merge now.  Thanks!